pkg/api/customization/kontainerdriver/actionhandler.go

```diff
@@ -70,3 +83,51 @@ func (a ActionHandler) setDriverActiveStatus(apiContext *types.APIContext, statu
 
 	return err
 }
+
+func (lh ListHandler) LinkHandler(apiContext *types.APIContext, next types.RequestHandler) error {
+	k8sCurr := strings.Split(settings.KubernetesVersionsCurrent.Get(), ",")
+	rkeSysImages, winSysImages := map[string]v3.RKESystemImages{}, map[string]v3.WindowsSystemImages{}
+	for _, k8sVersion := range k8sCurr {
+		rkeSysImg, err := kd.GetRKESystemImages(k8sVersion, lh.SysImageLister, lh.SysImages)
+		if err != nil {
+			return err
+		}
+		rkeSysImages[k8sVersion] = rkeSysImg
+		// ui allows only >= 1.14 for windows
+		k8sMajor := util.GetTagMajorVersion(k8sVersion)
+		if k8sMajor <= "v1.13" {
+			continue
+		}
+		winSysImg, err := kd.GetRKEWindowsSystemImages(k8sVersion, lh.WinSysImageLister, lh.WinSysImages)
+		if err != nil {
+			return err
+		}
+		winSysImages[k8sVersion] = winSysImg
+	}
+
+	targetImages, err := image.CollectionImages(rkeSysImages, winSysImages)
+	if err != nil {
+		return err
+	}
+
+	b := []byte(strings.Join(targetImages, "\n"))
+	apiContext.Response.Header().Set("Content-Length", strconv.Itoa(len(b)))
+	apiContext.Response.Header().Set("Content-Type", "application/octet-stream")
+	apiContext.Response.Header().Set("Content-Disposition", fmt.Sprintf("attachment; filename=%s.txt", "rancher-images"))
+	apiContext.Response.Header().Set("Cache-Control", "private")
+	apiContext.Response.Header().Set("Pragma", "private")
+	apiContext.Response.Header().Set("Expires", "Wed 24 Feb 1982 18:42:00 GMT")
```

Comment:
Why do we add an Expires header?

Reply:
I think it's because browser doesn't cache the file